### ISSUE TITLE
Gradio API calls with NestJS

### ISSUE THREAD

**Issue by PhantHive** [ipsa-scrypt/sushi#3]
**Describe the bug**

**To Reproduce**
1. Create a NestJS API (commonjs)
2. Call gradio api using their package @gradio/client

**Expected behavior**
Return the response from the AI

**Logs**
Errors are either: 

* eventsource not recognized
```sh
ReferenceError: EventSource is not defined 
at file:///F:/DOSSIER%20ULTIME/PROG/01-JS/Framework/reactJS/sushi/node_modules/@gradio/client/dist/index.js:734:13 
at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**We can work around the above error:**
```ts
const EventSource = require('eventsource');
globalThis.EventSource = EventSource;
```

* url not defined 
```sh
TypeError: The "url" argument must be of type string. 
Received an instance of URL at new NodeError
(node:internal/errors:406:5) at validateString (node:internal/validators:162:11) at Url.parse (node:url:170:3) at urlParse (node:url:141:13) at connect
 (F:\DOSSIER ULTIME\PROG\01-JS\Framework\reactJS\sushi\node_modules\eventsource\lib\eventsource.js:90:19) 
at new EventSource (F:\DOSSIER ULTIME\PROG\01-JS\Framework\reactJS\sushi\node_modules\eventsource\lib\eventsource.js:283:3) 
at file:///F:/DOSSIER%20ULTIME/PROG/01-JS/Framework/reactJS/sushi/node_modules/@gradio/client/dist/index.js:734:27
at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**We can work around the above error:**
In @gradio/client/dist/index.js
```ts
 eventSource = new EventSource(url.toString());
 ```
 Instead of:
```ts
 eventSource = new EventSource(url);
```

**Additional context**
All of the above are not proper solutions to solve the issue.

**Comment by PhantHive** [comment]
I will try with Gradio 4.0.0 tonight with express.

**Comment by PhantHive** [comment]
Temporary fixed with a script `modifyGradio.js` to make changes in the @gradio module.
PS: this script will be included during the deploy process.

[Fixed in this commit](881389af0e9478856d8cb61bd7cd7849ba159d50)
**Gradio: 4.7.1**
**@gradio/client**: 0.8.1
